mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -2058,32 +2058,36 @@ def Vector_GatherOp :
     Results<(outs AnyVectorOfNonZeroRank:$result)> {
 
   let summary = [{
-    gathers elements from memory or ranked tensor into a vector as defined by an
-    index vector and a mask vector
+    Gathers elements from memory or ranked tensor into a vector as defined by an
+    index vector and a mask vector.
   }];
 
   let description = [{
     The gather operation returns an n-D vector whose elements are either loaded
-    from memory or ranked tensor, or taken from a pass-through vector, depending
+    from a k-D memref or tensor, or taken from an n-D pass-through vector, depending
     on the values of an n-D mask vector.
-    If a mask bit is set, the corresponding result element is defined by the base
-    with indices and the n-D index vector (each index is a 1-D offset on the base).
-    Otherwise, the corresponding element is taken from the n-D pass-through vector.
-    Informally the semantics are:
+
+    If a mask bit is set, the corresponding result element is taken from `base`
+    at an index defined by k `indices` and n-D `index_vec`. Otherwise, the element
+    is taken from the pass-through vector. As an example, suppose that base is
+    3-D and the result is 2-D. The indexing semantics are then,
```

Comment:
I think it would be helpful to show the gather we are talking about and all the ssa values `mask`, `base`, `indices`, `pass_thru`, with their data types